virttest/migration_template.py

```diff
@@ -586,17 +727,40 @@ def _set_vm_disk(self, cache="none"):
         :param cache: vm disk cache mode
         """
         logging.debug("Prepare shared disk in vm xml for live migration")
-        if self.storage_type:
-            if self.storage_type == 'nfs':
-                logging.debug("Prepare nfs backed disk in vm xml")
-                for vm in self.vms:
-                    libvirt.update_vm_disk_source(vm.name,
-                                                  self.nfs_mount_dir)
-                    libvirt.update_vm_disk_driver_cache(vm.name,
-                                                        driver_cache=cache)
-            else:
-                # TODO:other storage types
-                self.test.cancel("Only nfs storage is supported for now")
+        if self.storage_type == 'nfs':
```

Comment:
I observed more and more methods are introduced to this template class as more requirements is coming. Eventually it will becomes one big bang class, which will make it harder to be readable,and maintainable.
I think whether we can split them into a few class,such as storagetemplate.class, networktemplate.class, then use migration_template inherit them.

Reply:
The methods in migration_template.py are all well designed for good readability and easy maintenance. For migration feature itself, there are only two categories in this aspect: migration with storage copy, migration without storage copy. Other features like network/cpu/etc won't be handled in migration_template.py.